components/Theme/index.tsx

```diff
@@ -1,9 +1,32 @@
+'use client';
+
+import { setCookie } from 'cookies-next';
+
+import { useTheme } from '@/context/ThemeProvider';
+import { ModeType } from '@/types';
+
 import MoonIcon from './MoonIcon';
+import SunIcon from './SunIcon';
 
 const ThemeSwitch = () => {
+  const { mode, setMode } = useTheme();
+
+  const setTheme = (theme: ModeType) => {
+    const themeCookie = theme === 'light' ? 'dark' : 'light';
+    setCookie('theme', themeCookie);
+    setMode(theme);
```

Comment:
```suggestion
    setCookie('theme', theme);
    setMode(theme);
```
You are already toggling the theme in your `onClick` function, so we don't need to toggle again here.